File: jdk/Step06/src/hasa/Gun.java
package hasa;

public class Gun {
	protected int bullet;

	public Gun() {
		bullet = 20;
	}
	
	public void shot() {
		if(bullet>0) {
			bullet--;
			System.out.println("BBang!!");
		}else {
			reload();
		}
	}

	public void reload() {
		System.out.println("reload reload!");
		bullet = 20;
	}
	
	
}

File: C#/Step04/Step04/Gun.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Step04
{
    public abstract class Gun
    {
        public abstract void shot();
        public virtual void reload()
        {
            Console.WriteLine("reload!!");
        }
    }

    class M16 : Gun
    {
        public override void shot()
        {
            Console.WriteLine("BBang!!");
        }
        
    }
}
